File: 剑指offer/easy/20210914/isStraight.cpp
class Solution {
public:
    bool isStraight(vector<int>& nums) {
        int Min=14,Max=-1;
        map<int,int>hash;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==0){
                continue;
            }
            if(hash[nums[i]]!=0)
            return false;
            hash[nums[i]]=1;
            Max=max(Max,nums[i]);
            Min=min(Min,nums[i]);
        }
        return Max-Min<5;
    }
};

File: Leetcode/Review/20211004/isStraight.go
package _0211004

import "math"

func isStraight(nums []int) bool {
	max,min:=math.MinInt32,math.MaxInt32
	hash:=map[int]int{}
	for i:=0;i<len(nums);i++{
		hash[nums[i]]++
		if (hash[nums[i]]>1&&nums[i]!=0){
			return false
		}
		if nums[i]>max{
			max=nums[i]
		}
		if nums[i]<min&&nums[i]!=0{
			min=nums[i]
		}
	}
	return max-min<5
}